WooCommerce/Classes/ViewModels/CardPresentPayments/CardPresentModalUpdateProgress.swift

```diff
@@ -0,0 +1,82 @@
+import UIKit
+
+/// Modal presented when a firmware update is being installed
+///
+final class CardPresentModalUpdateProgress: CardPresentPaymentsModalViewModel {
+    /// Amount of update that has been completed
+    private let progress: Float
+
+    /// Called when cancel button is tapped
+    private let cancelAction: (() -> Void)?
+
+    let textMode: PaymentsModalTextMode = .fullInfo
+    let actionsMode: PaymentsModalActionsMode
+
+    var topTitle: String
+
+    var topSubtitle: String? = nil
+
+    let image: UIImage
+
+    let primaryButtonTitle: String? = nil
+
+    let secondaryButtonTitle: String? = Localization.cancel
+
+    let auxiliaryButtonTitle: String? = nil
+
+    let bottomTitle: String?
+
+    var bottomSubtitle: String? = Localization.message
+
+    init(progress: Float, cancel: (() -> Void)?) {
+        self.progress = progress
+        self.cancelAction = cancel
+
+        topTitle = progress == 1 ? Localization.titleComplete : Localization.title
```

Comment:
Comparing floats to ints makes me nervous. Could we do this differently? e.g. progress > 0.99 or something?

Reply:
(Resolved below, at least for now :) )